### PR TITLE
Support for storage resultSet in Parquet and Orc format

### PR DESCRIPTION
**Pull request by ChengJie1053**
Support for storing result sets in Parquet format

`linkis.engine.resultSet.type` default value is `dolphin`, Can be modified to `parquet` and `orc`

The maven compilation command is `mvn clean install -Pstorage-parquet  -Pstorage-orc  -Dmaven.test.skip=true`

### CODE REVIEW

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/conf/LinkisStorageConf.java [hunk 1]
```diff
@@ -34,12 +40,17 @@ public class LinkisStorageConf {
   public static final String ROW_BYTE_MAX_LEN_STR =
       CommonVars.apply("wds.linkis.resultset.row.max.str", "2m").getValue();
 
+  public static final String ENGINE_RESULT_TYPE =
+      CommonVars.apply("wds.linkis.engine.resultSet.type", DOLPHIN, "Result type").getValue();
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/conf/LinkisStorageConf.java:44]: need remote wds prefix

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/DefaultResultSetFactory.java [hunk 2]
```diff
@@ -134,15 +135,22 @@ public String[] getResultSetType() {
 
   @Override
   public ResultSet<? extends MetaData, ? extends Record> getResultSetByPath(FsPath fsPath, Fs fs) {
+    ResultSet resultSet = null;
     try (InputStream inputStream = fs.read(fsPath)) {
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/DefaultResultSetFactory.java:139]: should support filename suffix judge

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ParquetResultSetReader<K extends MetaData, V extends Record>
+    extends StorageResultSetReader {
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java:45]: should extends ResultSetReader

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+  @Override
+  public Record getRecord() {
+    if (metaData == null) throw new RuntimeException("Must read metadata first(必须先读取metadata)");
+    if (row == null) {
+      throw new RuntimeException(
+          "Can't get the value of the field, maybe the IO stream has been read or has been closed!(拿不到字段的值，也许IO流已读取完毕或已被关闭！)");
+    }
+    try {
+      this.record = parquetReader.read();
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java:110]: this.record = parquetReader.read(); should in hasNext method

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetWriter.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ParquetResultSetWriter<K extends MetaData, V extends Record>
+    extends StorageResultSetWriter {
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetWriter.java:44]: should extends ResultSetWriter

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetWriter.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    }
+  }
+
+  @Override
+  public void addRecord(Record record) {
+    if (moveToWriteRow) {
+      try {
+        TableRecord tableRecord = (TableRecord) record;
+        Object[] row = tableRecord.row;
+        try (ParquetWriter<GenericRecord> writer =
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetWriter.java:77]: writer should be class var, only init once

Comment (ChengJie1053) [reply linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetWriter.java]: Ok, thanks for reviewing the code

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/OrcResultSetReader.java [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+      } catch (Exception e) {
+        throw new RuntimeException("Failed to read parquet schema", e);
+      }
+    }
+    return metaData;
+  }
+
+  @Override
+  public int skip(int recordNum) throws IOException {
+    return 0;
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/OrcResultSetReader.java:104]: need to implement

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/OrcResultSetReader.java [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+  }
+
+  @Override
+  public long getPosition() throws IOException {
+    return 0;
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/OrcResultSetReader.java:109]: need to implement

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/OrcResultSetReader.java [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+  }
+
+  @Override
+  public long available() throws IOException {
+    return 0;
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/OrcResultSetReader.java:114]: need to implement

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+      } catch (Exception e) {
+        throw new RuntimeException("Failed to read parquet schema", e);
+      }
+    }
+    return metaData;
+  }
+
+  @Override
+  public int skip(int recordNum) throws IOException {
+    return 0;
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java:101]: need to implement

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  @Override
+  public long getPosition() throws IOException {
+    return 0;
+  }
+
+  @Override
+  public long available() throws IOException {
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java:110]: need to implement

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  @Override
+  public int skip(int recordNum) throws IOException {
+    return 0;
+  }
+
+  @Override
+  public long getPosition() throws IOException {
+    return 0;
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ParquetResultSetReader.java:106]: need to implement

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ResultSetReaderFactory.java [hunk 13]
```diff
@@ -40,8 +42,24 @@ public class ResultSetReaderFactory {
...
-    return new StorageResultSetReader<>(resultSet, inputStream);
+      ResultSet<K, V> resultSet, InputStream inputStream, FsPath fsPath) {
+    ResultSetReader<K, V> resultSetReader = null;
+    if (fsPath.getPath().endsWith(Dolphin.DOLPHIN_FILE_SUFFIX)) {
+      resultSetReader = new StorageResultSetReader<>(resultSet, inputStream);
+    } else if (fsPath.getPath().endsWith(LinkisStorageConf.PARQUET_FILE_SUFFIX)) {
+      try {
+        resultSetReader = new ParquetResultSetReader<>(resultSet, inputStream, fsPath);
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to read parquet", e);
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/ResultSetReaderFactory.java:53]: Use Linkis Exception

File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/source/FileSource.java [hunk 14]
```diff
@@ -73,7 +73,9 @@ public interface FileSource extends Closeable {
       (path, suffix) -> path.endsWith("." + suffix);
 
   static boolean isResultSet(String path) {
-    return suffixPredicate.apply(path, fileType[0]);
+    return suffixPredicate.apply(path, fileType[0])
```
Comment (peacewong) [inline linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/source/FileSource.java:76]: need to update dolphin

File: linkis-public-enhancements/linkis-pes-publicservice/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java [hunk 15]
```diff
@@ -404,6 +404,10 @@ public Message getDirFileTrees(
     FsPathListWithError fsPathListWithError = fileSystem.listPathWithError(fsPath);
     if (fsPathListWithError != null) {
       for (FsPath children : fsPathListWithError.getFsPaths()) {
+        // 兼容parquet和orc,跳过.crc文件
```
Comment (peacewong) [inline linkis-public-enhancements/linkis-pes-publicservice/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java:407]: need use English

Comment (ChengJie1053) [reply linkis-public-enhancements/linkis-pes-publicservice/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java:407]: Ok, thanks for reviewing the code

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**